98_exercises.Rmd

```diff
@@ -291,24 +291,87 @@ See the OMA book for examples.
 
 ### MultiAssayExperiment (MAE) data container
 
-1. Create TreeSE data containers from individual CSV files.
-2. Combine TreeSE into MAE.
-3. Check that each individual experiment of MAE equals corresponding TreeSE.
-4. Take a subset of MAE (e.g., 10 first samples), and observe the subsetted MAE.
-
-Useful functions: DataFrame, TreeSummarizedExperiment, matrix, rownames, colnames, MultiAssayExperiment, ExperimentList, SimpleList
-
+The following exercise walks you through disassembling a MAE object in
+order to store it as multiple separate csv files.
+
+1. Import the microbiomeDataSets package
+2. Store the HintikkaXOData MAE into a variable named `mae`
+3. Which and how many experiments are contained by the MAE?
+4. Split the MAE into TreeSEs named `microbiota`, `metabolites` and `biomarkers`
+5. Which and how many assays are contained by each TreeSE?
+6. Store the assays of each TreeSE into a variable named `<TreeSE name>_assays`
+7. Write a csv file for each assays object with the `write.csv` function
+8. Besides experiments, MAEs also include a sampleData and a sampleMap, which
+   are accessible with `colData(mae)` and `sampleMap(mae)`, respectively.
+   Save also each of these two elements into a csv file.
+
+Well done! You just splitted a MAE into its components and stored them as csv files.
+[This script](https://github.com/JuliaTurkuDataScience/MicrobiomeAnalysis.jl/blob/main/src/assets/XO_preprocess.R)
```

Comment:
This refers to Julia code. Is this good for R tutorial?

Reply:
[XO_preprocess.R](https://github.com/JuliaTurkuDataScience/MicrobiomeAnalysis.jl/blob/main/src/assets/XO_preprocess.R) is an R script and is basically the solution to the exercise.

But if you think it is confusing because it is in a Julia package, then I'll remove it (I don't think anyone would notice though).

Reply:
I think we should not mix R and Julia material this way. 

Or what would be the argument for keeping that R script in Julia package, instead of OMA repository or mia R package repository?

Reply:
Because I used it in the documentation to show how to disassemble MAEs in R and import them in Julia: https://juliaturkudatascience.github.io/MicrobiomeAnalysis.jl/dev/example3/#Importing-MAE

Reply:
Would there be a more suitable place in OMA or mia?